core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/main/dot-edit-page-nav/dot-edit-page-nav.component.spec.ts

```diff
@@ -53,62 +76,55 @@ describe('DotEditPageNavComponent', () => {
-                            snapshot: {
-                                firstChild: {
-                                    url: [
-                                        {
-                                            path: 'content'
-                                        }
-                                    ]
-                                }
-                            }
-                        }
-                    }
-                ]
-            });
+    beforeEach(waitForAsync(() => {
+        TestBed.configureTestingModule({
+            imports: [RouterTestingModule, TooltipModule, DotIconModule, DotPipesModule],
+            declarations: [DotEditPageNavComponent, TestHostComponent],
+            providers: [
+                { provide: DotMessageService, useValue: messageServiceMock },
+                { provide: DotLicenseService, useClass: MockDotLicenseService },
+                { provide: DotPropertiesService, useClass: MockDotPropertiesService },
+                {
+                    provide: DotContentletEditorService,
+                    useClass: MockDotContentletEditorService
+                },
+                {
+                    provide: ActivatedRoute,
+                    useClass: ActivatedRouteMock
+                }
+            ]
+        });
 
-            fixture = TestBed.createComponent(TestHostComponent);
-            de = fixture.debugElement;
-            component = de.query(By.css('dot-edit-page-nav')).componentInstance;
-            fixture.componentInstance.pageState = new DotPageRenderState(
-                mockUser(),
-                new DotPageRender(mockDotRenderedPage())
-            );
-            dotContentletEditorService = TestBed.inject(DotContentletEditorService);
-            fixture.detectChanges();
-        })
-    );
+        fixture = TestBed.createComponent(TestHostComponent);
+        de = fixture.debugElement;
+        component = de.query(By.css('dot-edit-page-nav')).componentInstance;
+        fixture.componentInstance.pageState = new DotPageRenderState(
+            mockUser(),
+            new DotPageRender(mockDotRenderedPage())
+        );
+        dotContentletEditorService = TestBed.inject(DotContentletEditorService);
+        route = TestBed.inject(ActivatedRoute);
+    }));
 
     describe('basic setup', () => {
         it('should have menu list', () => {
+            fixture.detectChanges();
             const menuList = fixture.debugElement.query(By.css('.edit-page-nav'));
             expect(menuList).not.toBeNull();
         });
 
         it('should have correct item active', () => {
+            fixture.detectChanges();
             const activeItem = fixture.debugElement.query(By.css('.edit-page-nav__item--active'));
             expect(activeItem.nativeElement.innerText).toContain('CONTENT');
         });
 
         it('should call the ContentletEditorService Edit when clicked on Properties button', () => {
+            fixture.detectChanges();
```

Comment:
I did that in that way because when I use spyOnProperty at `should not has Experiments item` I need to do the spy before call the `fixture.detectChanges();`